File: backend/src/main/java/com/example/Flicktionary/domain/movie/dto/MovieResponse.java
package com.example.Flicktionary.domain.movie.dto;

import com.example.Flicktionary.domain.movie.entity.Movie;
import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.lang.NonNull;

@AllArgsConstructor
@Getter
public class MovieResponse {
    @NonNull
    private final Long id;
    @NonNull
    private final String title;
    private final String posterPath;
    @NonNull
    private final double averageRating;
    @NonNull
    private final int ratingCount;

    public MovieResponse(Movie movie) {
        this(movie.getId(), movie.getTitle(), movie.getPosterPath(), movie.getAverageRating(), movie.getRatingCount());
    }
}

File: backend/src/main/kotlin/com/example/Flicktionary/domain/movie/dto/MovieResponse.kt
package com.example.Flicktionary.domain.movie.dto

import com.example.Flicktionary.domain.movie.entity.Movie

data class MovieResponse(
    val id: Long,
    val title: String,
    val posterPath: String?,
    val averageRating: Double,
    val ratingCount: Int
) {
    constructor(movie: Movie) : this(
        movie.id,
        movie.title,
        movie.posterPath,
        movie.averageRating,
        movie.ratingCount
    )
}
